components/logging/org.wso2.carbon.logging.appender.http/src/main/java/org/wso2/carbon/logging/appender/http/utils/queue/PersistentQueue.java

```diff
@@ -0,0 +1,242 @@
+                    "Error: Unable to calculate disk usage.", e);
+        }
+    }
+
+    // appender should be loaded first to allow equality check when tailer is being loaded
+    private void loadAppenderBlock() throws PersistentQueueException {
+
+        JsonArray queueBlocks = this.queueMetaDataHandler.getAsJsonArray(QUEUE_BLOCK_LIST_KEY);
+        String appenderBlockName = queueBlocks.get(queueBlocks.size() - 1).getAsString();
+        this.appenderBlock = QueueBlock.loadBlock(this.queueDirectoryPath, appenderBlockName);
+    }
+
+    private void loadTailerBlock() throws PersistentQueueException {
+
+        JsonArray queueBlocks = this.queueMetaDataHandler.getAsJsonArray(QUEUE_BLOCK_LIST_KEY);
+        String tailerBlockName = queueBlocks.get(0).getAsString();
+        if(appenderBlock.getFileName().equals(tailerBlockName)) {
+            this.tailerBlock = appenderBlock;
+        }
+        else {
+            this.tailerBlock = QueueBlock.loadBlock(this.queueDirectoryPath, tailerBlockName);
+        }
+    }
+
+    private byte[] serializeObject(T obj) throws PersistentQueueException {
+
+        ByteArrayOutputStream byteArrayOutputStream = new ByteArrayOutputStream();
+        try (ObjectOutputStream objectOutputStream = new ObjectOutputStream(
+                new GZIPOutputStream(byteArrayOutputStream))) {
+            objectOutputStream.writeObject(obj);
+        }
+        catch (IOException e) {
+            throw new PersistentQueueException(
+                    PersistentQueueException.PersistentQueueErrorTypes.QUEUE_MESSAGE_SERIALIZATION_FAILED,
+                    "Error while serializing object", e);
+        }
+        return byteArrayOutputStream.toByteArray();
+    }
+
+    @SuppressWarnings("unchecked")
+    private T deserializeObject(byte[] bytes) throws PersistentQueueException {
+
+        try (ObjectInputStream objectInputStream = new ObjectInputStream(
+                new GZIPInputStream(new ByteArrayInputStream(bytes)))) {
+            return (T)objectInputStream.readObject();
+        }
+        catch (IOException | ClassNotFoundException e) {
+            throw new PersistentQueueException(
+                    PersistentQueueException.PersistentQueueErrorTypes.QUEUE_MESSAGE_DESERIALIZATION_FAILED,
+                    "Error while deserializing object", e);
+        }
+    }
+
+    private QueueBlock createNewBlock() throws PersistentQueueException {
+
+        if((currentDiskUsage + maxBatchSizeInBytes) > maxDiskSpaceInBytes) {
+            throw new PersistentQueueException(
+                    PersistentQueueException.PersistentQueueErrorTypes.QUEUE_DISK_SPACE_LIMIT_EXCEEDED,
+                    "Error: Queue disk usage limit reached.");
+        }
+        String QUEUE_BLOCK_FILE_NAME = "qb_%s";
+        String newBlockName = String.format(QUEUE_BLOCK_FILE_NAME, System.currentTimeMillis());
+        QueueBlock newBlock = new QueueBlock(this.queueDirectoryPath, newBlockName, maxBatchSizeInBytes);
```

Comment:
no need to pass block size to check file creation

Reply:
The block size here s needed to set the file size of the queue block. Therefore needs to be passed.